src/NuGet.Clients/NuGet.PackageManagement.UI/UserAction.cs

```diff
@@ -26,8 +26,8 @@ private UserAction(NuGetProjectActionType action, string packageId, NuGetVersion
             ActiveTab = activeTab;
         }
 
-        private UserAction(NuGetProjectActionType action, string packageId, NuGetVersion? packageVersion, bool isSolutionLevel, ContractsItemFilter activeTab, VersionRange versionRange)
-            : this(action, packageId, packageVersion, isSolutionLevel, activeTab)
+        private UserAction(NuGetProjectActionType action, string packageId, NuGetVersion? packageVersion, bool isSolutionLevel, ContractsItemFilter activeTab, VersionRange versionRange, string? sourceMappingSourceName)
+            : this(action, packageId, packageVersion, isSolutionLevel, activeTab, sourceMappingSourceName)
```

Comment:
naming: SourceName?

It's basicallyt he source the package got installed from or is it more?

Reply:
The source that the UI passed into the action, which is the selected source in the dropdown.
Maybe `selectedSourceName` or `selectedSourceNameForSourceMapping`  ?

Reply:
I like the first one. 2nd seems verbose.